File: extensions/CData ADO.NET Provider for MongoDB 2015/demos - winform/querybrowser/cs/ConnectionStringForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Resources;
using System.Data.CData.MongoDB;
using System.Data.Common;
using System.Threading;
using System.Reflection;

namespace CData.MongoDB.Demo {
  public class ConnectionStringForm : Form {
      /// <summary>
    /// Required designer variable.
    /// </summary>
    private System.ComponentModel.IContainer components = null;
    private string CONNECTED_SUCCESSFULLY;
    private string CONNECTION_NAME_EMPTY;

    /// <summary>
    /// Clean up any resources being used.
    /// </summary>
    /// <param name="disposing">true if managed resources should be disposed; otherwise, false.</param>
    protected override void Dispose(bool disposing) {
      if (disposing && (components != null)) {
        components.Dispose();
      }
      base.Dispose(disposing);
    }

    /// <summary>
    /// Required method for Designer support - do not modify
    /// the contents of this method with the code editor.
    /// </summary>     
    /// 
    private void InitializeComponent() {
      System.ComponentModel.ComponentResourceManager resources = new System.ComponentModel.ComponentResourceManager(typeof(ConnectionStringForm));
      this.connectionSettingLabel = new System.Windows.Forms.Label();
      this.settingPropertyGrid = new System.Windows.Forms.PropertyGrid();
      this.cancelButton = new System.Windows.Forms.Button();
      this.okButton = new System.Windows.Forms.Button();
      this.testbutton = new System.Windows.Forms.Button();
      this.connectionNameLabel = new System.Windows.Forms.Label();
      this.connectionNameText = new System.Windows.Forms.TextBox();
      this.SuspendLayout();
      // 
      // connectionSettingLabel
      // 
      this.connectionSettingLabel.AutoSize = true;
      this.connectionSettingLabel.Location = new System.Drawing.Point(12, 9);
      this.connectionSettingLabel.Name = "connectionSettingLabel";
      this.connectionSettingLabel.Size = new System.Drawing.Size(191, 13);
      this.connectionSettingLabel.TabIndex = 17;
      this.connectionSettingLabel.Text = resources.GetString("ConnectionSettingLabel_Text");
      // 
      // settingPropertyGrid
      // 
      this.settingPropertyGrid.Anchor = ((System.Windows.Forms.AnchorStyles)((((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Bottom)
                  | System.Windows.Forms.AnchorStyles.Left)
                  | System.Windows.Forms.AnchorStyles.Right)));
      this.settingPropertyGrid.Location = new System.Drawing.Point(15, 77);
      this.settingPropertyGrid.Name = "settingPropertyGrid";
      this.settingPropertyGrid.Size = new System.Drawing.Size(405, 366);
      this.settingPropertyGrid.TabIndex = 2;
      // 
      // cancelButton
      // 
      this.cancelButton.Anchor = ((System.Windows.Forms.AnchorStyles)((System.Windows.Forms.AnchorStyles.Bottom | System.Windows.Forms.AnchorStyles.Right)));
      this.cancelButton.DialogResult = System.Windows.Forms.DialogResult.Cancel;
      this.cancelButton.Location = new System.Drawing.Point(345, 449);
      this.cancelButton.Name = "cancelButton";
      this.cancelButton.Size = new System.Drawing.Size(75, 23);
      this.cancelButton.TabIndex = 5;
      this.cancelButton.UseVisualStyleBackColor = true;
      this.cancelButton.Text = resources.GetString("CancelButton_Text");
      // 
      // okButton
      // 
      this.okButton.Anchor = ((System.Windows.Forms.AnchorStyles)((System.Windows.Forms.AnchorStyles.Bottom | System.Windows.Forms.AnchorStyles.Right)));
      this.okButton.DialogResult = System.Windows.Forms.DialogResult.OK;
      this.okButton.Location = new System.Drawing.Point(264, 449);
      this.okButton.Name = "okButton";
      this.okButton.Size = new System.Drawing.Size(75, 23);
      this.okButton.TabIndex = 4;
      this.okButton.Text = "OK";
      this.okButton.UseVisualStyleBackColor = true;
      this.okButton.Click += new System.EventHandler(this.OnOkButtonClick);
      // 
      // testbutton
      // 
      this.testbutton.Anchor = ((System.Windows.Forms.AnchorStyles)((System.Windows.Forms.AnchorStyles.Bottom | System.Windows.Forms.AnchorStyles.Left)));
      this.testbutton.Location = new System.Drawing.Point(15, 449);
      this.testbutton.Name = "testbutton";
      this.testbutton.Size = new System.Drawing.Size(103, 23);
      this.testbutton.TabIndex = 3;
      this.testbutton.UseVisualStyleBackColor = true;
      this.testbutton.Click += new System.EventHandler(this.OnTestButtonClick);
      this.testbutton.Text = resources.GetString("Testbutton_Text");
      // 
      // connectionNameLabel
      // 
      this.connectionNameLabel.AutoSize = true;
      this.connectionNameLabel.Location = new System.Drawing.Point(12, 34);
      this.connectionNameLabel.Name = "connectionNameLabel";
      this.connectionNameLabel.Size = new System.Drawing.Size(95, 13);
      this.connectionNameLabel.TabIndex = 22;
      this.connectionNameLabel.Text = resources.GetString("ConnectionNameLabel_Text");
      // 
      // connectionNameText
      // 
      this.connectionNameText.Location = new System.Drawing.Point(15, 51);
      this.connectionNameText.Name = "connectionNameText";
      this.connectionNameText.Size = new System.Drawing.Size(405, 20);
      this.connectionNameText.TabIndex = 1;
      // 
      // ConnectionStringForm
      //
      this.AutoScaleDimensions = new System.Drawing.SizeF(6F, 13F);
      this.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font;
      this.ClientSize = new System.Drawing.Size(432, 484);
      this.Controls.Add(this.connectionNameText);
      this.Controls.Add(this.connectionNameLabel);
      this.Controls.Add(this.testbutton);
      this.Controls.Add(this.okButton);
      this.Controls.Add(this.cancelButton);
      this.Controls.Add(this.settingPropertyGrid);
      this.Controls.Add(this.connectionSettingLabel);
      this.DoubleBuffered = true;
      this.MaximizeBox = false;
      this.MinimizeBox = false;
      this.Name = "ConnectionStringForm";
      this.ShowIcon = false;
      this.ShowInTaskbar = false;
      this.SizeGripStyle = System.Windows.Forms.SizeGripStyle.Show;
      this.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent;
      this.Load += new System.EventHandler(this.OnSettingsFormLoad);
      this.FormClosed += new System.Windows.Forms.FormClosedEventHandler(this.OnSettingsFormClosed);
      this.ResumeLayout(false);
      this.PerformLayout();
      this.Text = resources.GetString("ConnectionStringForm_Text");

      CONNECTED_SUCCESSFULLY = resources.GetString("CONNECTED_SUCCESSFULLY");
      CONNECTION_NAME_EMPTY = resources.GetString("CONNECTION_NAME_EMPTY");
    }

    private System.Windows.Forms.Label connectionSettingLabel;
    private System.Windows.Forms.PropertyGrid settingPropertyGrid;
    private System.Windows.Forms.Button cancelButton;
    private System.Windows.Forms.Button okButton;
    private System.Windows.Forms.Button testbutton;
   
    private MongoDBConnectionStringBuilder _builder = null;
    private Thread _testConnectionTread = null;
    public string connectionString = null;
    public string connectionName = null;
    private Label connectionNameLabel;
    private TextBox connectionNameText;
    private bool _aborted = false;
    private delegate void EnableControlDeletgate(Control control, bool enable);
    private delegate void SetCursorDelegate(Form form, Cursor cursor);

    public ConnectionStringForm(string connectionString, string name) {
      InitializeComponent();
      this._builder = new MyConnectionStringBuilder();
      this.connectionString = connectionString;
      this.connectionName = name;
    }

    private void TestConnection() {
      this.Invoke(new EnableControlDeletgate(DoEnableControl), this.settingPropertyGrid, false);
      this.Invoke(new EnableControlDeletgate(DoEnableControl), this.okButton, false);
      this.Invoke(new EnableControlDeletgate(DoEnableControl), this.testbutton, false);
      this.Invoke(new SetCursorDelegate(DoSetCursor), this, Cursors.WaitCursor);
      try {
        MongoDBConnection _conn = new MongoDBConnection(_builder.ConnectionString);
        if (_conn.State == ConnectionState.Closed) {
          _conn.Open();
        }
        _conn.TestConnection();
        _conn.Close();
        MessageBox.Show(CONNECTED_SUCCESSFULLY, this.Text, MessageBoxButtons.OK, MessageBoxIcon.Information, MessageBoxDefaultButton.Button1);
      } catch (Exception ex) {
        if (!this._aborted) {
          MessageBox.Show(ex.Message, this.Text, MessageBoxButtons.OK, MessageBoxIcon.Error, MessageBoxDefaultButton.Button1);
        }
        return;
      } finally {
        if (!this._aborted) {
          this.Invoke(new EnableControlDeletgate(DoEnableControl), this.settingPropertyGrid, true);
          this.Invoke(new EnableControlDeletgate(DoEnableControl), this.okButton, true);
          this.Invoke(new EnableControlDeletgate(DoEnableControl), this.testbutton, true);
          this.Invoke(new SetCursorDelegate(DoSetCursor), this, Cursors.Default);
        }
      }
      return;
    }

    private void DoEnableControl(Control control, bool enable) {
      control.Enabled = enable;
    }

    private void DoSetCursor(Form form, Cursor cursor) {
      form.Cursor = cursor;
    }

    private void OnSettingsFormLoad(object sender, EventArgs e) {
      if (!string.IsNullOrEmpty(this.connectionString)) {
        try {
          this._builder.ConnectionString = this.connectionString;
        } catch {
          this._builder.ConnectionString = string.Empty;
        }
      }
      if (!string.IsNullOrEmpty(this.connectionName)) {
        this.connectionNameText.Text = this.connectionName;
      }
      this.settingPropertyGrid.SelectedObject = _builder;
    }

    private void OnTestButtonClick(object sender, EventArgs e) {
      if (this._testConnectionTread != null && this._testConnectionTread.ThreadState == ThreadState.Running) {
        this._testConnectionTread.Abort();
      }
      this._testConnectionTread = new Thread(this.TestConnection);
      this._testConnectionTread.Start();
    }

    private void OnOkButtonClick(object sender, EventArgs e) {
      if (string.IsNullOrEmpty(this.connectionNameText.Text)) {
        MessageBox.Show(CONNECTION_NAME_EMPTY, null, MessageBoxButtons.OK, MessageBoxIcon.Error, MessageBoxDefaultButton.Button1);
      } else {
        this.connectionString = _builder.ConnectionString;
        this.connectionName = this.connectionNameText.Text;
        this.DialogResult = DialogResult.OK;
        this.Close();
      }
    }

    private void OnSettingsFormClosed(object sender, FormClosedEventArgs e) {
      if (this._testConnectionTread != null) {
        this._testConnectionTread.Abort();
        this._aborted = true;
      }
    }
  }

  [DefaultPropertyAttribute(null)]
  internal class MyConnectionStringBuilder : MongoDBConnectionStringBuilder {

  }
}

File: extensions/CData ADO.NET Provider for MongoDB 2015/demos - winform/querybrowser/vb/ConnectionStringForm.vb
Imports System.Collections.Generic
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports System.Data.CData.MongoDB
Imports System.Data.Common
Imports System.Resources
Imports System.Threading
Imports System.Reflection
Imports System.ComponentModel
Namespace CData.MongoDB.Demo
  Public Class ConnectionStringForm
    Inherits Form
    ''' <summary>
    ''' Required designer variable.
    ''' </summary>
    Private components As System.ComponentModel.IContainer = Nothing
    Private CONNECTED_SUCCESSFULLY As String = Nothing
    Private CONNECTION_NAME_EMPTY As String = Nothing
    ''' <summary>
    ''' Clean up any resources being used.
    ''' </summary>
    ''' <param name="disposing">true if managed resources should be disposed; otherwise, false.</param>
    Protected Overrides Sub Dispose(disposing As Boolean)
      If disposing AndAlso (components IsNot Nothing) Then
        components.Dispose()
      End If
      MyBase.Dispose(disposing)
    End Sub

    ''' <summary>
    ''' Required method for Designer support - do not modify
    ''' the contents of this method with the code editor.
    ''' </summary>     
    ''' 
    Private Sub InitializeComponent()
      Dim resources = New System.ComponentModel.ComponentResourceManager(GetType(ConnectionStringForm))
      Me.connectionSettingLabel = New System.Windows.Forms.Label()
      Me.settingPropertyGrid = New System.Windows.Forms.PropertyGrid()
      Me.cancelButton = New System.Windows.Forms.Button()
      Me.okButton = New System.Windows.Forms.Button()
      Me.testbutton = New System.Windows.Forms.Button()
      Me.connectionNameLabel = New System.Windows.Forms.Label()
      Me.connectionNameText = New System.Windows.Forms.TextBox()
      Me.SuspendLayout()
      ' 
      ' connectionSettingLabel
      ' 
      Me.connectionSettingLabel.AutoSize = True
      Me.connectionSettingLabel.Location = New System.Drawing.Point(12, 9)
      Me.connectionSettingLabel.Name = "connectionSettingLabel"
      Me.connectionSettingLabel.Size = New System.Drawing.Size(185, 13)
      Me.connectionSettingLabel.TabIndex = 17
      Me.connectionSettingLabel.Text = resources.GetString("ConnectionSettingLabel_Text")
      ' 
      ' settingPropertyGrid
      ' 
      Me.settingPropertyGrid.Anchor = CType((((System.Windows.Forms.AnchorStyles.Top Or System.Windows.Forms.AnchorStyles.Bottom) Or System.Windows.Forms.AnchorStyles.Left) Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles)
      Me.settingPropertyGrid.Location = New System.Drawing.Point(15, 77)
      Me.settingPropertyGrid.Name = "settingPropertyGrid"
      Me.settingPropertyGrid.Size = New System.Drawing.Size(405, 366)
      Me.settingPropertyGrid.TabIndex = 2
      ' 
      ' cancelButton
      ' 
      Me.cancelButton.Anchor = CType((System.Windows.Forms.AnchorStyles.Bottom Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles)
      Me.cancelButton.DialogResult = System.Windows.Forms.DialogResult.Cancel
      Me.cancelButton.Location = New System.Drawing.Point(345, 449)
      Me.cancelButton.Name = "cancelButton"
      Me.cancelButton.Size = New System.Drawing.Size(76, 23)
      Me.cancelButton.TabIndex = 5
      Me.cancelButton.UseVisualStyleBackColor = True
      Me.cancelButton.Text = resources.GetString("CancelButton_Text")
      ' 
      ' okButton
      ' 
      Me.okButton.Anchor = CType((System.Windows.Forms.AnchorStyles.Bottom Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles)
      Me.okButton.DialogResult = System.Windows.Forms.DialogResult.OK
      Me.okButton.Location = New System.Drawing.Point(264, 449)
      Me.okButton.Name = "okButton"
      Me.okButton.Size = New System.Drawing.Size(76, 23)
      Me.okButton.TabIndex = 4
      Me.okButton.Text = "OK"
      Me.okButton.UseVisualStyleBackColor = True
      AddHandler Me.okButton.Click, New System.EventHandler(AddressOf Me.OnOkButtonClick)
      ' 
      ' testbutton
      ' 
      Me.testbutton.Anchor = CType((System.Windows.Forms.AnchorStyles.Bottom Or System.Windows.Forms.AnchorStyles.Left), System.Windows.Forms.AnchorStyles)
      Me.testbutton.Location = New System.Drawing.Point(15, 449)
      Me.testbutton.Name = "testbutton"
      Me.testbutton.Size = New System.Drawing.Size(103, 23)
      Me.testbutton.TabIndex = 3
      Me.testbutton.UseVisualStyleBackColor = True
      AddHandler Me.testbutton.Click, New System.EventHandler(AddressOf Me.OnTestButtonClick)
      Me.testbutton.Text = resources.GetString("Testbutton_Text")
      ' 
      ' connectionNameLabel
      ' 
      Me.connectionNameLabel.AutoSize = True
      Me.connectionNameLabel.Location = New System.Drawing.Point(12, 34)
      Me.connectionNameLabel.Name = "connectionNameLabel"
      Me.connectionNameLabel.Size = New System.Drawing.Size(95, 13)
      Me.connectionNameLabel.TabIndex = 22
      Me.connectionNameLabel.Text = resources.GetString("ConnectionNameLabel_Text")
      ' 
      ' connectionNameText
      ' 
      Me.connectionNameText.Location = New System.Drawing.Point(15, 51)
      Me.connectionNameText.Name = "connectionNameText"
      Me.connectionNameText.Size = New System.Drawing.Size(405, 20)
      Me.connectionNameText.TabIndex = 1
      ' 
      ' ConnectionStringForm
      ' 
      Me.AutoScaleDimensions = New System.Drawing.SizeF(6F, 13F)
      Me.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font
      Me.ClientSize = New System.Drawing.Size(432, 484)
      Me.Controls.Add(Me.connectionNameText)
      Me.Controls.Add(Me.connectionNameLabel)
      Me.Controls.Add(Me.testbutton)
      Me.Controls.Add(Me.okButton)
      Me.Controls.Add(Me.cancelButton)
      Me.Controls.Add(Me.settingPropertyGrid)
      Me.Controls.Add(Me.connectionSettingLabel)
      Me.DoubleBuffered = True
      Me.MaximizeBox = False
      Me.MinimizeBox = False
      Me.Name = "ConnectionStringForm"
      Me.ShowIcon = False
      Me.ShowInTaskbar = False
      Me.SizeGripStyle = System.Windows.Forms.SizeGripStyle.Show
      Me.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent
      AddHandler Me.FormClosed, New System.Windows.Forms.FormClosedEventHandler(AddressOf Me.OnSettingsFormClosed)
      AddHandler Me.Load, New System.EventHandler(AddressOf Me.OnSettingsFormLoad)
      Me.ResumeLayout(False)
      Me.PerformLayout()
      Me.Text = resources.GetString("ConnectionStringForm_Text")
      Me.CONNECTED_SUCCESSFULLY = resources.GetString("CONNECTED_SUCCESSFULLY")
      Me.CONNECTION_NAME_EMPTY = resources.GetString("CONNECTION_NAME_EMPTY")
    End Sub
    Private connectionSettingLabel As System.Windows.Forms.Label
    Private settingPropertyGrid As System.Windows.Forms.PropertyGrid
    Private cancelButton As System.Windows.Forms.Button
    Private okButton As System.Windows.Forms.Button
    Private testbutton As System.Windows.Forms.Button

    Private _builder As MongoDBConnectionStringBuilder = Nothing
    Private _testConnectionTread As Thread = Nothing
    Public connectionString As String = Nothing
    Public connectionName As String = Nothing
    Private connectionNameLabel As Label
    Private connectionNameText As TextBox
    Private _aborted As Boolean = False
    Private Delegate Sub EnableControlDeletgate(control As Control, enable As Boolean)
    Private Delegate Sub SetCursorDelegate(form As Form, cursor As Cursor)

    Public Sub New(connectionString As String, name As String)
      InitializeComponent()
      Me._builder = New MyConnectionStringBuilder()
      Me.connectionString = connectionString
      Me.connectionName = name
    End Sub

    Private Sub TestConnection()
      Me.Invoke(New EnableControlDeletgate(AddressOf DoEnableControl), Me.settingPropertyGrid, False)
      Me.Invoke(New EnableControlDeletgate(AddressOf DoEnableControl), Me.okButton, False)
      Me.Invoke(New EnableControlDeletgate(AddressOf DoEnableControl), Me.testbutton, False)
      Me.Invoke(New SetCursorDelegate(AddressOf DoSetCursor), Me, Cursors.WaitCursor)
      Try
        Dim _conn As New MongoDBConnection(_builder.ConnectionString)
        If _conn.State = ConnectionState.Closed Then
          _conn.Open()
        End If
        _conn.TestConnection()
        _conn.Close()
        MessageBox.Show(CONNECTED_SUCCESSFULLY, Me.Text, MessageBoxButtons.OK, MessageBoxIcon.Information, MessageBoxDefaultButton.Button1)
      Catch ex As Exception
        If Not Me._aborted Then
          MessageBox.Show(ex.Message, Me.Text, MessageBoxButtons.OK, MessageBoxIcon.[Error], MessageBoxDefaultButton.Button1)
        End If
        Return
      Finally
        If Not Me._aborted Then
          Me.Invoke(New EnableControlDeletgate(AddressOf DoEnableControl), Me.settingPropertyGrid, True)
          Me.Invoke(New EnableControlDeletgate(AddressOf DoEnableControl), Me.okButton, True)
          Me.Invoke(New EnableControlDeletgate(AddressOf DoEnableControl), Me.testbutton, True)
          Me.Invoke(New SetCursorDelegate(AddressOf DoSetCursor), Me, Cursors.[Default])
        End If
      End Try
      Return
    End Sub

    Private Sub DoEnableControl(control As Control, enable As Boolean)
      control.Enabled = enable
    End Sub

    Private Sub DoSetCursor(form As Form, cursor As Cursor)
      form.Cursor = cursor
    End Sub

    Private Sub OnSettingsFormLoad(sender As Object, e As EventArgs)
      If Not String.IsNullOrEmpty(Me.connectionString) Then
        Try
          Me._builder.ConnectionString = Me.connectionString
        Catch
          Me._builder.ConnectionString = String.Empty
        End Try
      End If
      If Not String.IsNullOrEmpty(Me.connectionName) Then
        Me.connectionNameText.Text = Me.connectionName
      End If
      Me.settingPropertyGrid.SelectedObject = _builder
    End Sub

    Private Sub OnTestButtonClick(sender As Object, e As EventArgs)
      If Me._testConnectionTread IsNot Nothing AndAlso Me._testConnectionTread.ThreadState = ThreadState.Running Then
        Me._testConnectionTread.Abort()
      End If
      Me._testConnectionTread = New Thread(AddressOf Me.TestConnection)
      Me._testConnectionTread.Start()
    End Sub

    Private Sub OnOkButtonClick(sender As Object, e As EventArgs)
      If String.IsNullOrEmpty(Me.connectionNameText.Text) Then
        MessageBox.Show(CONNECTION_NAME_EMPTY, Nothing, MessageBoxButtons.OK, MessageBoxIcon.[Error], MessageBoxDefaultButton.Button1)
      Else
        Me.connectionString = _builder.ConnectionString
        Me.connectionName = Me.connectionNameText.Text
        Me.DialogResult = DialogResult.OK
        Me.Close()
      End If
    End Sub

    Private Sub OnSettingsFormClosed(sender As Object, e As FormClosedEventArgs)
      If Me._testConnectionTread IsNot Nothing Then
        Me._testConnectionTread.Abort()
        Me._aborted = True
      End If
    End Sub
  End Class

  <DefaultPropertyAttribute(Nothing)> _
  Friend Class MyConnectionStringBuilder
    Inherits MongoDBConnectionStringBuilder

  End Class
End Namespace